src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -170,34 +186,67 @@ void icm426xxGyroInit(gyroDev_t *gyro)
     spiWriteReg(dev, ICM426XX_RA_PWR_MGMT0, ICM426XX_PWR_MGMT0_TEMP_DISABLE_OFF | ICM426XX_PWR_MGMT0_ACCEL_MODE_LN | ICM426XX_PWR_MGMT0_GYRO_MODE_LN);
     delay(15);
 
-    uint8_t outputDataRate = 0;
+    uint8_t odrIdx = 0;
     bool supportedODRFound = false;
 
     if (gyro->gyroRateKHz) {
-        uint8_t gyroSyncDenominator = gyro->mpuDividerDrops + 1; // rebuild it in here, see gyro_sync.c
-        uint8_t desiredODRKhz = 8 / gyroSyncDenominator;
-        for (uint32_t i = 0; i < ARRAYLEN(icm426xxPkhzToSupportedODRMap); i++) {
-            if (icm426xxPkhzToSupportedODRMap[i].khz == desiredODRKhz) {
-                outputDataRate = icm426xxPkhzToSupportedODRMap[i].odr;
+        const uint8_t gyroSyncDenominator = gyro->mpuDividerDrops + 1; // rebuild it in here, see gyro_sync.c
+        const uint8_t desiredODR = 8 / gyroSyncDenominator;
+        for (uint32_t i = 0; i < ARRAYLEN(odrConfigLUT); i++) {
```

Comment:
```suggestion
        for (uint8_t i = 0; i < ARRAYLEN(odrConfigLUT); i++) {
```

Reply:
`ARRAYLEN` uses the `sizeof` operator which returns a 32bit value, afaik. With `i` being `uint8_t`, maybe there is an additional cast to `uint32_t` needed when comparing it to `ARRAYLEN(odrConfigLUT)`. What do other people think about this?

Reply:
Or event better: `unsigned i`

Reply:
https://godbolt.org/z/eYj3b8j7W

Reply:
That's what I expected. So uint32_t can stay.

Reply:
But when reading code, `unsigned` means 'some reasonably sized unsigned integer value'. `uint32_t` looks like 32bits are needed for some reason (which is not this case). `size_t` somewhere between the cases above, explicitly referring to some object size (array in this case).
Personally, I'd use `unsigned` here (and it is commonly used in BF), but `size_t` is acceptable.

BTW: both clang and gcc generate extra `UXTx`/`SXTx` instruction on almost each operation with `[u]int16`/`[u]int8_t` types. So IMO it is better to use them only when variable must be stored in memory (static/global/struct)